miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -739,7 +739,7 @@ private void applyTissueProcessingChanges(SampleTissueProcessing target, SampleT
    */
   private boolean isExternalNameDuplicatedInProject(Sample sample) throws IOException {
     SampleIdentity identity = (SampleIdentity) sample;
-    return getIdentitiesByExactExternalNameAndProject(identity.getExternalName(), identity.getProject().getId()).size() > 0;
+    return getIdentitiesByExternalNameOrAliasAndProject(identity.getExternalName(), identity.getProject().getId(), true).size() > 0;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)